hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/AsyncCleanerService.java

```diff
@@ -43,7 +43,7 @@ public class AsyncCleanerService extends HoodieAsyncTableService {
 
   protected AsyncCleanerService(BaseHoodieWriteClient writeClient) {
     super(writeClient.getConfig());
-    this.writeClient = writeClient;
+    this.writeClient = writeClient.createNewClient();
```

Comment:
even if the fs view got updated, the cleaner should also work because it only cares about history commits/file slice version, so this fix might be unnecessary(Flink always uses the async cleaning in ingestion pipeline and we never got any issue there).